### PR TITLE
fix breakage on Arch by using Debian libtinfo5_6.0 to satisfy clang

### PR DESCRIPTION
**Pull request by superbaud**
This builds off of @defuse's https://github.com/zcash/zcash/pull/5553 and closes https://github.com/zcash/zcash/issues/5488.

i tested it on an arch machine and a debian machine, both seem to build fine and the `native_libtinfo` depend doesn't get used on the debian machine, as intended.

### CODE REVIEW

**Review by daira — APPROVED** [review]
utACK

**Review by defuse — APPROVED** [review]

**Comment by str4d** [comment]
I would have preferred that the two commits were squashed into a single commit. In particular, this PR being opened separately from #5553 (instead of building on it) without squashing the commits, mean that GitHub showed #5553 as having been merged with no review and no passing tests, which was a red flag to me until I discovered this PR.

File: depends/packages/packages.mk [hunk 1]
```diff
@@ -1,6 +1,10 @@
 zcash_packages := libsodium utfcpp
 packages := boost libevent zeromq $(zcash_packages) googletest
-native_packages := native_clang native_ccache native_rust
+native_packages := native_clang native_ccache native_rust
```
Comment (str4d) [inline depends/packages/packages.mk:3]: The change to this line should have been undone.
```suggestion
native_packages := native_clang native_ccache native_rust
```

**Review by str4d — COMMENTED** [review]
Post-hoc utACK

**Comment by superbaud** [comment]
> I would have preferred that the two commits were squashed into a single commit. In particular, this PR being opened separately from #5553 (instead of building on it) without squashing the commits, mean that GitHub showed #5553 as having been merged with no review and no passing tests, which was a red flag to me until I discovered this PR.

That makes sense and thanks for pointing that out. I'll squash the commits next time and also figure out how to amend someone else's PR (I think by doing a PR of my own to *their* branch?) next time I'm in this sorta situation.

File: depends/packages/packages.mk (hunk 1 above)
Comment (superbaud) [reply depends/packages/packages.mk:3]: thanks for spotting that; i can't reopen this PR (tried both with the web interface and the `gh` command line interface) so i'm going to submit a new PR fixing this

Comment (superbaud) [reply depends/packages/packages.mk:3]: https://github.com/zcash/zcash/pull/5600

**Pull request merged**